src/ragger/conftest/base_conftest.py

```diff
@@ -0,0 +1,205 @@
+]
+
+
+def pytest_addoption(parser):
+    parser.addoption("--device", choices=DEVICES, required=True)
+    parser.addoption("--backend", choices=BACKENDS, default="speculos")
+    parser.addoption("--display", action="store_true", default=False)
+    parser.addoption("--golden_run", action="store_true", default=False)
+    parser.addoption("--log_apdu_file", action="store", default=None)
+
+
+@pytest.fixture(scope="session")
+def backend_name(pytestconfig):
+    return pytestconfig.getoption("backend")
+
+
+@pytest.fixture(scope="session")
+def display(pytestconfig):
+    return pytestconfig.getoption("display")
+
+
+@pytest.fixture(scope="session")
+def golden_run(pytestconfig):
+    return pytestconfig.getoption("golden_run")
+
+
+@pytest.fixture(scope="session")
+def log_apdu_file(pytestconfig):
+    filename = pytestconfig.getoption("log_apdu_file")
+    return Path(filename).resolve() if filename is not None else None
+
+
+@pytest.fixture(scope="session")
+def root_pytest_dir(request):
+    return Path(request.config.rootpath).resolve()
+
+
+@pytest.fixture
+def test_name(request):
+    # Get the name of current pytest test
+    test_name = request.node.name
+
+    # Remove firmware suffix:
+    # -  test_xxx_transaction_ok[nanox 2.0.2]
+    # => test_xxx_transaction_ok
+    return test_name.split("[")[0]
+
+
+# Glue to call every test that depends on the firmware once for each required firmware
+def pytest_generate_tests(metafunc):
+    if "firmware" in metafunc.fixturenames:
+        fw_list = []
+        ids = []
+
+        device = metafunc.config.getoption("device")
+        backend_name = metafunc.config.getoption("backend")
+
+        if device == "all":
+            if backend_name != "speculos":
+                raise ValueError("Invalid device parameter on this backend")
+
+            # Add all supported firmwares
+            for fw in FIRMWARES:
+                fw_list.append(fw)
+                ids.append(fw.device + " " + fw.version)
+
+        else:
+            # Enable firmware for demanded device
+            for fw in FIRMWARES:
+                if device == fw.device:
+                    fw_list.append(fw)
+                    ids.append(fw.device + " " + fw.version)
+
+        metafunc.parametrize("firmware", fw_list, ids=ids, scope="session")
+
+
+def prepare_speculos_args(root_pytest_dir: Path, firmware: Firmware, display: bool):
+    speculos_args = []
+
+    if display:
+        speculos_args += ["--display", "qt"]
+
+    device = firmware.device
+    if device == "nanosp":
+        device = "nanos2"
+
+    # Find the compiled application for the requested device
+    project_root_dir = find_project_root_dir(root_pytest_dir)
+
+    app_path = Path(project_root_dir / "build" / device / "bin" / "app.elf").resolve()
+    if not app_path.is_file():
+        raise ValueError(f"File '{app_path}' missing. Did you compile for this target?")
+
+    if not len(conf.OPTIONAL["SIDELOADED_APPS"]) == 0:
```

Comment:
```suggestion
    if conf.OPTIONAL["SIDELOADED_APPS"]:
```
this is equivalent no?

Reply:
It's not as resilient for example when messing with `None` values or if a boolean is inserted
I find the idea of checking the length of the list more clear too